vendor/github.com/bitrise-io/stepman/cli/preload_steps.go

```diff
@@ -0,0 +1,269 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/bitrise-io/go-utils/colorstring"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/stepman/models"
+	"github.com/bitrise-io/stepman/stepman"
+)
+
+const (
+	bitriseStepLibURL = "https://github.com/bitrise-io/bitrise-steplib.git"
+	bitriseMaintainer = "bitrise"
+	workers           = 10
+)
+
+type GoBuilder func(stepSourceAbsPath, packageName, targetExecutablePath string) error
```

Comment:
This whole file is heavily coupled with the Go step toolkit while stepman should be neutral. I would rather expend the step toolkit interface with the required methods and hooks to make this code uncoupled from any single toolkit.

Reply:
Like the idea of expanding the toolkit interface.
However we can not use the toolkit interface here directly, as it would result in circular dependency.
Can look into removing the Go specificness.